Makefile

```diff
@@ -35,6 +35,7 @@ PWD ?= `pwd`
 GIT ?= git
 TELEPORT_DEBUG ?= false
 GITTAG=v$(VERSION)
+GITREF=$(shell git describe --tags --long --match "v[0-9]*")
```

Comment:
There is a corner case here that I encountered in another workflow. If another tag is created that points to the same commit then this command will pull the new tag rather that one that started the build. For example, if a tag build for `v1.2.3` is started and I create and push `v1.2.3-fred.1` before this line is ran, `git describe --tags --long --match "v[0-9]*"` will produce a value based off of my `v1.2.3-fred.1` tag.

Reply:
Oh, man. This is pain to fix because `git describe --match` takes a glob, not a regex. To make it work for out use case, we would need something like this to be able to handle the potential 2-digit numbers:
```
git describe --tags --long --match "v1[0-9].[0-9].[0-9]"  --match "v1[0-9].[0-9][0-9].[0-9]"  --match "v1[0-9].[0-9].[0-9][0-9]"  --match "v1[0-9].[0-9][0-9].[0-9][0-9]"
```

Reply:
I am currently using `git describe --tags --match "v[[:digit:]]*.[[:digit:]]*.[[:digit:]]"` in a few places, but I'm not 100% sure that it is correct. Globs are not fun outside of trivial use cases and it seems like everywhere they are implemented is _slightly_ different...

Reply:
@fheinecke - that glob is better, but I think it still doesn't protect us from something like `v1.2.3-fred.1`, as it ends with `.1` 😞.

Reply:
I suppose we can always just `--exclude "*-*"`. It would mean getting slightly stranger tags for dev builds, though.